packages/transformers/src/lit-transformer.ts

```diff
@@ -0,0 +1,398 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import * as ts from 'typescript';
+import {LitClassContext} from './lit-class-context.js';
+import {LitFileContext} from './lit-file-context.js';
+import {cloneNode} from 'ts-clone-node';
+
+import type {
+  Visitor,
+  ClassDecoratorVisitor,
+  MemberDecoratorVisitor,
+  GenericVisitor,
+} from './visitor.js';
+
+const unreachable = (x: never) => x;
+
+/**
+ * Configurable transformer for Lit.
```

Comment:
by "coalesce visitors" I mean that it's possible sometimes to write a batching visitor that calls visit on each of a set of visitors so they all go in one walk, rather than a series of walks. I think your code does this already.